js/app.recommendations.js

```
@@ -344,48 +486,66 @@
       const picked = [];
       const pickedKeys = new Set();
       const pickScheme = (scheme) => {
         picked.push(scheme);
         pickedKeys.add(scheme.schemeKey);
-        if (scheme.selectedMatchNames) {
-          scheme.selectedMatchNames.forEach((name) => remaining.delete(name));
+        const names = Array.isArray(scheme.effectiveSelectedMatchNames)
+          ? scheme.effectiveSelectedMatchNames
+          : scheme.selectedMatchNames;
+        if (Array.isArray(names)) {
+          names.forEach((name) => remaining.delete(name));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)